src/package/polyfill/overrideCustomElementDefine.ts

```diff
@@ -19,7 +24,8 @@ export function overrideCustomElementDefine() {
             if (!registeredCustomElement) {
                 const hookClass = new Proxy(createHookClass(elementName, impl), {
                     construct: function(element, args, newTarget) {
-                        return constructInstance(elementName, args, newTarget);
+                        const mostRecentImpl = getMostRecentImpl(elementName);
```

Comment:
needs to be change, this will calls the constructor instance that calls the classHook, that need the cache to be ready when it runs

to something like this:
```ts
           const registeredCustomElement = customElements.get(elementName);
          setMostRecentImpl(elementName, impl);
            if (!registeredCustomElement) {
                const hookClass = new Proxy(createHookClass(elementName, impl), {
                    construct: function(element, args, newTarget) {
                        const mostRecentImpl = getMostRecentImpl(elementName);                       
                        return constructInstance(mostRecentImpl, args, newTarget);
                    }
                });
                originalDefineFn.apply(this, [elementName, hookClass, options]);
            } else {

```

Reply:
just update code in review.. copy/paste error